File: app/src/main/java/com/easy/kotlins/http/core/OkCallback.java
package com.easy.kotlins.http.core;


import androidx.annotation.NonNull;

/**
 * Create by LiZhanPing on 2020/4/29
 */
public interface OkCallback<T> {

    void onSuccess(@NonNull T result);

    void onError(Throwable error);

}

File: common-ktx/src/main/java/com/easy/kotlins/http/core/OkCallback.kt
package com.easy.kotlins.http.core

/**
 * Create by LiZhanPing on 2020/4/29
 */
interface OkCallback<T> {
    fun onStart()
    fun onSuccess(result: T)
    fun onError(error: OkException)
    fun onComplete()
}